File: domain/bin/main/com/ahn/domain/usecase/GetSupportedCurrenciesUseCase.kt
package com.ahn.domain.usecase

import com.ahn.domain.model.CurrencyInfo
import com.ahn.domain.repository.ExchangeRateRepository
import javax.inject.Inject

class GetSupportedCurrenciesUseCase @Inject constructor(
    private val repository: ExchangeRateRepository
) {
    suspend operator fun invoke(): List<CurrencyInfo> {
        return repository.getSupportedCurrencies()
    }
}

File: ios/Packages/Domain/Sources/Domain/Exchange/UseCase/GetSupportedCurrenciesUseCase.swift
public struct GetSupportedCurrenciesUseCase: Sendable {
    private let repository: any ExchangeRateRepository

    public init(repository: any ExchangeRateRepository) {
        self.repository = repository
    }

    public func callAsFunction() async throws -> [CurrencyInfo] {
        try await repository.supportedCurrencies()
    }
}
